### PR TITLE
Add missing unsubscribe in SimpleConsumer, Addresses #305

### PR DESCRIPTION
**Pull request by colprog**

### CODE REVIEW

File: csharp/rocketmq-client-csharp/MetadataConstants.cs [hunk 1]
```diff
@@ -37,5 +38,10 @@ public class MetadataConstants
         public const string REQUEST_ID_KEY = "x-mq-request-id";
 
         public const string CLIENT_ID_KEY = "x-mq-client-id";
+
+        public static readonly string CLIENT_VERSION = Assembly
```
Comment (aaron-ai) [inline csharp/rocketmq-client-csharp/MetadataConstants.cs:42]: Nice! we shoud try to get version from predefined one from csproj.

File: csharp/rocketmq-client-csharp/rocketmq-client-csharp.csproj [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <PackageId>RocketMQ-Client-CSharp</PackageId>
-    <Version>1.0.0</Version>
+    <Version>5.0.0</Version>
```
Comment (aaron-ai) [inline csharp/rocketmq-client-csharp/rocketmq-client-csharp.csproj:5]:
```suggestion
    <Version>5.0.1</Version>

```

File: csharp/rocketmq-client-csharp/Client.cs [hunk 3]
```diff
@@ -518,11 +508,16 @@ internal virtual void OnSettingsReceived(rmq::Settings settings)
 
         protected readonly IClientManager Manager;
 
-        protected readonly HashSet<string> _topicsOfInterest = new HashSet<string>();
+        protected readonly ConcurrentDictionary<string, bool> _topicsOfInterest = new ();
```
Comment (aaron-ai) [inline csharp/rocketmq-client-csharp/Client.cs:511]: Why do we need a map here?

Comment (colprog) [reply csharp/rocketmq-client-csharp/Client.cs:511]: With this change, we will support dynamically subscribe/unsubscribe to multiple topics, maybe from different threads. C# does not have a ConcurrentSet, thus a ConcurrentDictionary to emulate one.

Comment (aaron-ai) [reply csharp/rocketmq-client-csharp/Client.cs:511]: OK

**Comment by aaron-ai** [comment]
Thanks @colprog !

**Pull request merged**